src/Sulu/Bundle/WebsiteBundle/Cache/CacheClearer.php

```diff
@@ -80,24 +80,27 @@ public function clear(/*?array $tags = null*/)
 
         $tags = \func_num_args() >= 1 ? \func_get_arg(0) : null;
 
-        if ($this->cacheManager && $this->cacheManager->supportsInvalidate()) {
-            if (null !== $tags && $this->cacheManager->supportsTags()) {
-                foreach ($tags as $tag) {
-                    $this->cacheManager->invalidateTag($tag);
-                }
-
-                $this->eventDispatcher->dispatch(new CacheClearEvent($tags), Events::CACHE_CLEAR);
-
-                return;
+        if (null !== $tags && $this->cacheManager && $this->cacheManager->supportsTags()) {
+            foreach ($tags as $tag) {
+                $this->cacheManager->invalidateTag($tag);
             }
 
-            $request = $this->requestStack->getCurrentRequest();
-            if (!$request) {
-                return;
-            }
+            $this->eventDispatcher->dispatch(new CacheClearEvent($tags), Events::CACHE_CLEAR);
+
+            return;
+        }
 
+        $request = $this->requestStack->getCurrentRequest();
+        if (null !== $request && $this->cacheManager && $this->cacheManager->supportsInvalidate()) {
             $this->cacheManager->invalidateDomain($request->getHost());
-            $this->eventDispatcher->dispatch(new CacheClearEvent($tags), Events::CACHE_CLEAR);
+            $this->eventDispatcher->dispatch(new CacheClearEvent(), Events::CACHE_CLEAR);
```

Comment:
we read the webspace from the tags to dispatch the `CacheCleared` domain event:
https://github.com/sulu/sulu/blob/0322de13b6c2860db6f0550d702b57acc1207392/src/Sulu/Bundle/WebsiteBundle/EventSubscriber/DomainEventEventSubscriber.php#L60-L63

if we dont pass any tag here, there will be no domain event 😕

Reply:
maybe we should adjust the `DomainEventEventSubscriber` to create a domain event for all webspaces, if no tags are set?